utils/handleError.ts

```diff
@@ -61,28 +34,44 @@ const FRONT_END_POSSIBLE_ERRORS = {
 
 type KnownFrontEndErrorTexts = keyof typeof FRONT_END_POSSIBLE_ERRORS;
 
-export function handleAuthErrors(e: unknown) {
-  if (isAxiosError(e) && e.response?.status === 400) {
-    if (typeof e.response.data !== 'string') {
-      console.error('Unexpected error occured!');
-      console.error('Unexpected error: ', e.response.data);
-      return 'Palvelin virhe!';
+export function handleError(e: unknown) {
+  if (e instanceof Error === false) {
+    console.error(e);
+    return 'Tuntematon virhe!';
+  }
+  console.error(e);
+  if (isAxiosError(e)) {
+    if (e.response?.status === 401) {
+      Router.push('/login').catch((e) => console.error(e));
+      return 'Istuntosi on vanhentunut! Ole hyvä ja kirjaudu uudelleen jatkaaksesi!';
     }
```

Comment:
Löysin Nextjs:stä jonkun tuommoisen `Router`in `import Router from 'next/router';`

Tuo näyttäisi pelittävän melkosamanlailla kuin `useRouter()` .tsx-tiedostoissa.

Ei harmainta aavistusta onko tuota hyvä käyttää, mutta näytti ainakin pelittävän :sweat_smile: 

En löytänyt dokumentaatiosta täysin vastaavasta `Router`ista tietoa, mutta tämä näyttäisi olevan lähimpänä: https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object

`Router`ia käyttämällä myös Toasti näkyy, missä lukee "Istuntosi on vanhentunut..."

Reply:
Tuota kutsutaan singleton routeriksi, ja on juuri oikea tapa tämmöiseen redirectiin 👍